src/core/sock/sock-redirect.h

```diff
@@ -85,6 +89,14 @@
 #include <mutex>
 #include <set>
 
+#ifdef XLIO_STATIC_BUILD
+#define DEF_SYM(_name)     xlio_##_name
+#define ORIG_SYMBOL(_name) _name
+#else
+#define DEF_SYM(_name)     _name
+#define ORIG_SYMBOL(_name) orig_os_api._name
+#endif /* XLIO_STATIC_BUILD */
```

Comment:
[suggestion] probably name with "SYSCALL" will be more readable. For example, glibc has something like this: `INTERNAL_SYSCALL_CALL`, `INLINE_SYSCALL`. With "INTERNAL_SYSCALL" or "XLIO_SYSCALL" we can call XLIO implementation and "EXTERNAL_SYSCALL" or "KERNEL_SYSCALL" will replace ORIG_SYMBOL.

Reply:
Also, need to choose between current format and the following one:
```C
#define DEF_SYM(_name, ...) xlio_##_name(__VA_ARGS__)
```
So, calling a syscall would look like DEF_SYM(close, fd)

Reply:
regarding naming: I like XLIO_CALL() more than XLIO(). In this case kernel syscall can follow this convention too - KERNEL_CALL(). Lets bring this discussion to wider forum.

Regarding VA_ARGS: these are just alternative options. I don't have strong opinion here. As for VA_ARGS option, this looks like commonly used approach and provides clear construction of function call, while `DEF_SYM(func)(args)` looks more complex.

Reply:
OK